tests/unit/regression/GaussianProcessRegression_unittest.cc

```diff
@@ -191,23 +191,24 @@ TEST(GaussianProcessRegression, apply_regression_on_training_features)
 	auto labels_train=std::make_shared<RegressionLabels>(lab_train);
 
 	// choose Gaussian kernel with sigma = 0.02 and zero mean function
-	auto kernel=std::make_shared<GaussianKernel>(10, 0.02);
-	auto mean=std::make_shared<ZeroMean>();
+	std::shared_ptr<Kernel> kernel=std::make_shared<GaussianKernel>(10, 0.02);
+	std::shared_ptr<MeanFunction> mean=std::make_shared<ZeroMean>();
 
 	// Gaussian likelihood with sigma = 0.25
-	auto liklihood=std::make_shared<GaussianLikelihood>(0.25);
+	std::shared_ptr<LikelihoodModel> liklihood=std::make_shared<GaussianLikelihood>(0.25);
 
 	// specify GP regression with exact inference
-	auto inf=std::make_shared<ExactInferenceMethod>(kernel, features_train,
-			mean, labels_train, liklihood);
-
+	auto inf=std::make_shared<ExactInferenceMethod>();
+	inf->put("kernel", kernel);
```

Comment:
could you use either string constants here (defined in the class you are putting to), or use c++ setters?